source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOIResourceResolver.h

```diff
@@ -9,6 +9,11 @@
 
 @protocol ACOIResourceResolver
 
+@optional
 - (UIImage *)resolveImageResource:(NSURL *)url;
+- (UIImage*) getRadioButtonForSelectedState;
```

Comment:
could we overload resolveImageResource:(NSURL *)url to take NSString and remove new interfaces. It seems too speicific. specifics of resource types being resolved is determined by scheme.

Reply:
I think this is more clean. As in later approach I will have to create 4 schema for 4 states.